open-grant-proposals/open-proposal-twitter-thread-rollup-backed-by-filecoin-and-nfts.md

```diff
@@ -0,0 +1,127 @@
+# Open Grant Proposal: `Twitter Thread Rollup backed by Filecoin and NFTs`
+
+**Name of Project:** NFTShapshot
+
+**Proposal Category:** `app-dev`, `docs`
+
+**Proposer:** `@jamesmorgan`
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** `Yes`
+
+# Project Description
+
+- Twitter thread rollup and timestamp bots are not a new thing, however what they currently do not utilise distributed, decentralised storage, such as the Filecoin network and then allow the creator to turn them into a tradeable NFTs (non-fungible tokens). 
+
+- Existing bots/solutions traditionally rely on centralised (hidden) storage and therefore limited in the what can be done with the thread that is snapshot'd.
+
+- The creation of a NFT derived from the timestamped twitter thread combined with a Filecoin stored rollup will bring two new things to the party:
+    - Facilitating trading of the thread - popular, viral or controversial threads may gain in value 
+    - Creating a source of revenue to maintain the tweeter bot and its Filecoin storage costs - we could possible charge $1-2 for creating the NFT as well as enabling trading on OpenSea with a secondary sales commission of 2% which will then go back into maintain the Filecoin storage
+    - Popularising Filecoin in general on NFT platforms - often NFTs use IPFS but in theory should be migrating to a 
+    Filecoin solution with a proper incentivisation layer - this would be a perfect candidate to test this out theory and popularise its usage
+
+## Value
+
+- What are the benefits to getting this right?
+
+* An example of how to use Filecoin in an NFT (popular future usecase) and an open source Github demo project
+* Promote the use of Filecoin in the twitter sphere in general
+* Provide a simple toolset that can be forked and used by many and demonstrates Filecoin
+
+- What are the risks if you don't get it right?
+
+* There may be reputational risk for the project if it fails, rendering the twitter bot unloved or sidelined.  
+
+- What are the risks that will make executing on this project difficult?
+
+I believe the technologies is not the hard part at least for ourselves, we have good experience in IPFS, building webapps and NFTs however there may be a problem in the incentivisation model in that supporting the file storage may become costly overtime. One option to avoid this is to snapshot the thread and store centrally, only pushing to filecoin when an NFT is created and the small fee is paid. 
+ 
+Filecoin is still emerging and the APIs may change in the short/medium term - keeping up to date may become probalatic but after review we think it is a fairly simply setup so this shouldn't be seen a huge risk right now.
+
```

Comment:
This projects sounds great. Wondering whether some of the Filecoin dev tools and libraries to make it easier for you to build this will be more ready in perhaps ~4-6 weeks. So if you were to start later in March you might be able to build this faster. Would starting later be ok?

Reply:
Later in March works for us, We want to hack something on Filecoin so very keen on the project and think its a good, easy to digest and accessible demonstration tool of the Filecoin protocol. Not sure what the next steps are but let me know if there is anything else you want us to do or if you want a call etc.